### PR TITLE
testlist/sim.dat: Skip 32bit sim build only for macOS

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
To fix the issue: https://github.com/apache/incubator-nuttx/issues/2094
Need merge https://github.com/apache/incubator-nuttx/pull/2128 first.

## Impact
The automation build

### CODE REVIEW

**Comment by btashton** [comment]
Can we consider what I did in this PR instead #71 I find to easier to have the restrictions for an arch in a single file rather than mixed with all the targets.

**Comment by xiaoxiang781216** [comment]
> Can we consider what I did in this PR instead #71 I find to easier to have the restrictions for an host in a single file rather than mixed with all the targets.

But puting all blacklist in one place give us the whole picture and the difference between host quickly.

> It would also make it easier to have modifiers a for a host added later.

I am thinking to change blacklist to regex pattern instead using config as regex, then we don't need change the match anymore.

File: testlist/other.dat [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# with the same name
+# /mips,CONFIG_MIPS32_TOOLCHAIN_PINGUINOL
+
+/risc-v,CONFIG_RV32IM_TOOLCHAIN_GNU_RVGL
+-gapuino:nsh
+-nr5m100-nexys4:nsh
+
+# x86_64-elf-gcc from homebrew doesn't seem to
+# provide __udivdi3 etc for -m32
+# /x86
```
Comment (btashton) [inline testlist/other.dat:24]: We should be able to re-enable this now with the Darwin exception.

File: testlist/sim.dat [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
--sim:sotest32
+-sim:loadable,Darwin
+-sim:module32,Darwin
+-sim:rpproxy,Darwin
+-sim:rpserver,Darwin
+-sim:sotest32,Darwin
 
 # X11
-# macOS doesn't have X11
+# macOS/Docker doesn't have X11
```
Comment (btashton) [inline testlist/sim.dat:13]: Not for this PR, but I'll get the X11 headers added to the docker image.

**Review by btashton — CHANGES_REQUESTED** [review]

File: testlist/sim.dat (hunk 2 above)
Comment (btashton) [reply testlist/sim.dat:13]: Dockerfile shows
```
  libx11-dev:i386 \
  libxext-dev:i386 \
```
I think these targets should build.

File: testlist/other.dat (hunk 1 above)
Comment (xiaoxiang781216) [reply testlist/other.dat]: Done, I move this change to another PR.

File: testlist/sim.dat (hunk 2 above)
Comment (xiaoxiang781216) [reply testlist/sim.dat]: So, I can enable it now?

Comment (xiaoxiang781216) [reply testlist/sim.dat]: Done.

**Comment by btashton** [comment]
Merged OS side, just restarted the build.

**Review by btashton — APPROVED** [review]

**Pull request merged**